### PR TITLE
Fix some startup issues in mixed mode

### PR DESCRIPTION
**Pull request by gacholio**
Some compile errors and heap settings.

[ci skip]

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

**Comment by gacholio** [comment]
Related: #643

File: runtime/gc_modron_startup/mminit.cpp [hunk 1]
```diff
@@ -1101,54 +1101,59 @@ gcInitializeXmxXmdxVerification(J9JavaVM *javaVM, IDATA* memoryParameters, bool
...
-	if (extensions->memoryMax > (extensions->heapCeiling - J9GC_COMPRESSED_POINTER_NULL_REGION_SIZE)) {
-		j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_GC_OPTION_OVERFLOW, displayXmxOrMaxRAMPercentage(memoryParameters));
-		return JNI_ERR;
+		if (extensions->memoryMax > (extensions->heapCeiling - J9GC_COMPRESSED_POINTER_NULL_REGION_SIZE)) {
+			j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_GC_OPTION_OVERFLOW, displayXmxOrMaxRAMPercentage(memoryParameters));
+			return JNI_ERR;
+		}
+	} else {
+		/* Constructor initialized heapCeiling to a compressed refs default - reset it if not compressed */
+		extensions->heapCeiling = 0;
```
Comment (DanHeidinga) [inline runtime/gc_modron_startup/mminit.cpp]: @charliegracie Can you comment on the approach here.  I'm not clear on whether there's a different (better?) way to do this.

**Comment by DanHeidinga** [comment]
Jenkins test sanity xlinux jdk8

File: runtime/gc_modron_startup/mminit.cpp (hunk 1 above)
Comment (gacholio) [reply runtime/gc_modron_startup/mminit.cpp]: Spoke with @charliegracie - I'll move the reset into the initializer.

Comment (gacholio) [reply runtime/gc_modron_startup/mminit.cpp]: This will unfortunately require the OMR change to happen first - marking WIP until promotion.

Comment (gacholio) [reply runtime/gc_modron_startup/mminit.cpp]: Actually doesn't require the OMR portion since we're not compiling mixed mode anywhere. Ready to merge.

**Comment by gacholio** [comment]
Jenkins test sanity xlinux jdk8

**Pull request merged**